enhancements/cloud-integration/infrastructure-external-platform-type.md

```diff
@@ -0,0 +1,513 @@
+2. Precisely define the reaction of the operators listed below for the "External" platform type
+3. Define and document concrete mechanism for supplement cluster with provider-specific components during installation time (CCM, MAPI controller, DNS controller)
+
+### Risks and Mitigations
+
+#### Support burden
+There is concern that *random* customers will use this feature out of context
+and create a support burden. However, using such platform type in conjunction with [capabilites selection](https://github.com/openshift/enhancements/blob/master/enhancements/installer/component-selection.md)
+could give us a clear signal about how to properly triage 3d party created replacements for Red Hat components.
+
+### Drawbacks
+
+Current approach with having statically defined platform list is already there, quite transparent and battle-hardened.
+Changing it by adding the new, way less specific platform type would mean a significant step away from current design patterns in this area.
+
+Also, future support strategy is not completely clear due to we have plans around enablement of 3d party platform-specific components, which
+codebase would be mostly out of our control.
+
+## Design Details
+
+### Open Questions
+
+- Should we explicitly communicate that the "External" platform is something that we do not support yet?
+- Should we gate the "External" platform addition behind the feature gate by generating separate CRD for TPNU clusters?
+  - Do we already have all the necessary mechanisms for gating such API changes behind the feature gate?
+- Should we invest in preparing workflows that will perform UPI with the "None"/"External" platform types installation
+  on the AWS or GCP, or existed vSphere-based workflows would be enough?
+
+### Test Plan
+
+During the first stages, we must ensure that OpenShift [built-in operators whose behaviour depends on platform type](#affected-components) treat the "External" platform type the same way as "None".
+To achieve this - existing infrastructure and mechanisms employed for exercising topologies with the "None" platform type might be used.
+
+For the moment of writing, the only workflow where the "None" platform tests is [upi-conf-vsphere-platform-none-ovn](https://steps.ci.openshift.org/workflow/upi-vsphere-platform-none-ovn).
+Based on this workflow, a new one with platform type "External" with a respective set of jobs should be added to ensure that we do not disrupt the current openshift operation.
+
+However, given that vSphere is the only platform where we're exercising clusters installation with platform "None" specified,
+it would be beneficial to develop additional workflows using a provider with better capacity and performance (AWS or GCP, perhaps).
+
+### Graduation Criteria
+
+Given that `infrastructure.config.openshift.io` has already been released and has to be supported during the whole major release lifecycle,
+this change will be GAed from the beginning.
```

Comment:
This should be updated based on whether we are TPing this or not

Reply:
Based on this duscussion: https://github.com/openshift/enhancements/pull/1234#discussion_r968935259
I think we have a consensus that we do not want to gate this with TPNU